protocol-units/execution/maptos/opt-executor/src/background/transaction_pipe.rs

```diff
@@ -305,20 +314,30 @@ impl TransactionPipe {
 		}
 
 		// Add the txn for future validation
-		let sequence_number = transaction.sequence_number();
-		debug!("Adding transaction to mempool: {:?} {:?}", transaction, sequence_number);
+		let state_view = self
+			.db_reader
+			.latest_state_checkpoint_view()
+			.expect("Failed to get latest state checkpoint view.");
+		let db_seq_num = get_account_sequence_number(&state_view, transaction.sender())?;
+		tracing::info!(
+			"Tx sender {} db seq number:{} tx seq num:{}",
+			transaction.sender(),
+			db_seq_num,
+			transaction.sequence_number()
+		);
+		let tx_hash = transaction.committed_hash();
 		let status = self.core_mempool.add_txn(
-			transaction.clone(),
+			transaction,
 			ranking_score,
-			sequence_number,
+			db_seq_num, //std::cmp::min(*db_seq_num + 1, sequence_number),
```

Comment:
Can you explain (best in the comments) why the number from the database is the correct one and it's appropriate to add the transaction with this number rather than reject and report an error, in case this number does not match transaction's intrinsic sequence number?

Reply:
Please remove the commented out code. And perhaps it's best to explain that the database sequence number is the right thing to use by aptos. Otherwise looks good.

Reply:
Agree, let's follow up as it's a detailed explain. And we are under the gun rn.